l10n_br_delivery_nfe/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2024 Diego Paradeda - KMEE
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockInvoiceOnshipping(models.TransientModel):
+    _inherit = "stock.invoice.onshipping"
+
+    vol_ids = fields.One2many(
+        string="Volume IDs Data",
+        comodel_name="stock.invoice.onshipping.vol",
+        inverse_name="invoice_wizard_id",
+    )
+
+    def _build_invoice_values_from_pickings(self, pickings):
+        invoice, values = super()._build_invoice_values_from_pickings(pickings)
+
+        # Volumes
+        vol_ids = self.vol_ids.filtered(lambda v: v.picking_id == pickings)
+        values["nfe40_vol"] = [
+            (
+                0,
+                0,
+                self._convert_volumes_to_write(vol_id),
+            )
+            for vol_id in vol_ids
+        ]
+
+        return invoice, values
+
+    def _convert_volumes_to_write(self, data):
+        """Prepare wizard lines for writing to fiscal document."""
+        return {
+            "nfe40_qVol": data.nfe40_qVol,
+            "nfe40_esp": data.nfe40_esp,
+            "nfe40_marca": data.nfe40_marca,
+            "nfe40_pesoL": data.nfe40_pesoL,
+            "nfe40_pesoB": data.nfe40_pesoB,
+        }
+
+    def _get_picking(self):
+        picking_ids = self._load_pickings()
+        if not picking_ids:
```

Comment:
Pelo o que entendi você criou esse método ao invés de usar o _load_pickings diretamente para incluir esse raise, é isso mesmo? Porque se for você pode colocar um TODO: Verificar na migração da v16 para passar a usar somente o _load_pickings, e na v16 acredito que não vai ter problema mover o código que está no action_generate_invoices para o _load_pickings

![image](https://github.com/OCA/l10n-brazil/assets/6341149/caefdadc-380c-418e-8e18-e02525157f82)

Isso parece ser uma simples refatoração que efetivamente não vai alterar nada porém torna esse método desnecessário

Reply:
De fato esse método não é necessário, vou remover